src/misc/programs.cpp

```diff
@@ -237,14 +237,14 @@ void Program::WriteOut(const char* format, ...)
 	truncated_chars_message(size);
 }
 
-void Program::WriteOut(const char* format, const char* arguments)
+void Program::WriteOut(const char* format, const std::string& arguments)
```

Comment:
You should really look at the usage code before making this change.  It's probably being called a lot with `WriteOut("blah", str.c_str())` and now it's doing a double conversion from `std::string` to `char *` and back again.

I would just leave this one alone personally.  There's a lot of C style string manipulation in the code.  Unless you're trying to do a full refactor to use C++ style string functions, there's not much point in changing this.

Reply:
Good point.

Reply:
Exactly @weirddan455. We don't get brownie points for using C++ string, usage matters 😄 if all you expect are string literals, `const char *` args are fine and sometimes even preferable.

Reply:
In this case, it was a `const char*` -> `std::string_view` -> `const std::string&` -> `const char*` transition. 😆 In the end, the good old `const char*` proved to be the best solution.

Your point is valid, especially in an environment like this, where const char* is still the norm in various parts of the project. Whether that is _always_ a good thing can be debated, since it lacks some safeties std::string/std:string_view have regarding string length, boundary checking, and such, but we can live with it.

Reply:
> Your point is valid, especially in an environment like this, where const char* is still the norm in various parts of the project. Whether that is _always_ a good thing can be debated, since it lacks some safeties std::string/std:string_view have regarding string length, boundary checking, and such, but we can live with it.

No doubt @japsmits, it would be nice to migrate everything to `std::string` in the codebase and use a logging library that uses `std::string` natively. But it would be a huge error-prone refactoring job with not too much of an upside. I wouldn't just switch to an entirely different logging library because of this.

So, this is the practical reality we live in now. Plus not sure how often strings are being passed back and forth between the core code and the various C/"minimal C++" libraries we're using.